src/azure-cli-core/azure/cli/core/commands/arm.py

```diff
@@ -730,3 +730,78 @@ def _find_property(instance, path):
     for part in path:
         instance = _update_instance(instance, part, path)
     return instance
+
+
+def assign_implict_identity(getter, setter, identity_role=None, identity_scope=None):
+    import time
+    from azure.mgmt.authorization import AuthorizationManagementClient
+    from azure.mgmt.authorization.models import RoleAssignmentProperties
+    from msrestazure.azure_exceptions import CloudError
+
+    # get
+    resource = getter()
+    if resource.identity:
+        logger.warning('Implict identity is already configured')
+    else:
+        resource = setter(resource)
+
+    # create role assignment:
+    if identity_scope:
+        principal_id = resource.identity.principal_id
+
+        identity_role_id = resolve_role_id(identity_role, identity_scope)
+        assignments_client = get_mgmt_service_client(AuthorizationManagementClient).role_assignments
+        properties = RoleAssignmentProperties(identity_role_id, principal_id)
+
+        logger.info("Creating an assignment with a role '%s' on the scope of '%s'", identity_role_id, identity_scope)
+        retry_times = 36
+        assignment_id = _gen_guid()
+        for l in range(0, retry_times):
+            try:
+                assignments_client.create(identity_scope, assignment_id, properties)
+                break
+            except CloudError as ex:
+                if 'role assignment already exists' in ex.message:
+                    logger.info('Role assignment already exists')
+                    break
+                elif l < retry_times and ' does not exist in the directory ' in ex.message:
+                    time.sleep(5)
+                    logger.warning('Retrying role assignment creation: %s/%s', l + 1,
+                                   retry_times)
+                    continue
+                else:
+                    raise
+    return resource
+
+
+def resolve_role_id(role, scope):
+    import uuid
+    from azure.mgmt.authorization import AuthorizationManagementClient
+    client = get_mgmt_service_client(AuthorizationManagementClient).role_definitions
+
+    role_id = None
+    if re.match(r'/subscriptions/.+/providers/Microsoft.Authorization/roleDefinitions/',
```

Comment:
The regex here can be improved because the wildcard is a GUID which is more restricted than `.+` matching.